app/misc/graphing.ts

```diff
@@ -10,15 +10,36 @@ let commitHistory = [];
 let commitList = [];
 let spacingY = 100;
 let spacingX = 80;
+let tagSpacingY = 100;
+let tagSpacingX = 300;
 let parentCount = {};
 let columns: boolean[] = [];
 let edgeDic = {};
 let numOfCommits = 0;
 let branchIds = {};
+let tagIds = {};
+let unumberPrev = 0;
+
+// In order to allow tags, branches, and nodes to have unique numerical id's
+// reference: https://stackoverflow.com/questions/8012002/create-a-unique-number-with-javascript-time
+function generateUniqueNumber() {
+    var date = Date.now();
+
+    // If created at same millisecond as previous
+    if (date <= unumberPrev) {
+        date = ++unumberPrev;
```

Comment:
So if it is <= we increment the value? Logically speaking this makes sense, since it is a date and it should be monotonically increasing; however theoretically, it is a bit confusing. Perhaps update the comment?